bamboolean/lexer.py

```diff
@@ -105,65 +107,51 @@ def _integer(self):
+        )
+
     def get_next_token(self):
         """
-        Lexical anaylzer (tokenizer). Breaks sentence apart into tokens
+        Lexical analyzer (tokenizer). Breaks sentence apart into tokens
         """
         while self.current_char is not None:
             if self.current_char.isspace():
                 self.skip_whitespace()
                 continue
 
-            if re.match('[_a-zA-Z]', self.current_char):
-                return self.id()
-
-            if self._is_quotation_mark(self.current_char):
-                self.next()
-                return self.string()
-
-            if self.current_char.isdigit():
-                return self.number()
-
-            # Relational operators
-
-            if self.current_char is '=' and self.peek() == '=':
-                self.next()
-                self.next()
-                return Token(tok.EQ, '==')
-
-            if self.current_char is '!' and self.peek() == '=':
-                self.next()
-                self.next()
-                return Token(tok.NE, '!=')
-
-            if self.current_char is '<':
-                # LTE, LT
-                self.next()
-                if self.current_char is '=':
-                    self.next()
-                    return Token(tok.LTE, '<=')
-                else:
-                    return Token(tok.LT, '<')
-
-            if self.current_char is '>':
-                # GTE, GT
-                self.next()
-                if self.current_char is '=':
-                    self.next()
-                    return Token(tok.GTE, '>=')
-                else:
-                    return Token(tok.GT, '>')
-
-            # Parentheses
-
-            if self.current_char == '(':
-                self.next()
-                return Token(tok.LPAREN, '(')
-
-            if self.current_char == ')':
-                self.next()
-                return Token(tok.RPAREN, ')')
-
+            regex_map = OrderedDict((
+                (r'("|\')', lambda: self.skip_and_invoke(1, self.string)),
+                (r'[_a-zA-Z]', self.id),
+                (r'\d', self.number),
+            ))
+
+            tokens_map = OrderedDict((
+                ('==', Token(tok.EQ, '==')),
+                ('!=', Token(tok.NE, '!=')),
+                ('<=', Token(tok.LTE, '<=')),
+                ('>=', Token(tok.GTE, '>=')),
+                ('>', Token(tok.GT, '>')),
+                ('<', Token(tok.LT, '<')),
+                ('(', Token(tok.LPAREN, '(')),
+                (')', Token(tok.RPAREN, ')')),
+            ))
+
+            for regex, func in regex_map.items():
+                if re.match(regex, self.current_char):
+                    return func()
+
+            for expected_val, token in tokens_map.items():
+                if self.is_token_equal(expected_val):
+                    return self.skip_and_invoke(len(expected_val),
+                                                lambda: token)
```

Comment:
`regex_map` and `tokens_map` probably can be merged into just one map. Although, i have not come up with straightforward solution that would not increase complexity, so decided not to waste more time on it. Suggestions are welcome.